src/providers/ApolloClientProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  ApolloClient,
+  ApolloProvider,
+  HttpLink,
+  InMemoryCache,
+  NormalizedCacheObject
+} from '@apollo/client'
+import { ConfigHelperConfig } from '@oceanprotocol/lib/dist/node/utils/ConfigHelper'
+import { useOcean } from '@oceanprotocol/react'
+import fetch from 'cross-fetch'
+import React, { useState, useEffect, ReactNode, ReactElement } from 'react'
+
+function createClient(subgraphUri: string) {
+  const client = new ApolloClient({
+    link: new HttpLink({
+      uri: `${subgraphUri}/subgraphs/name/oceanprotocol/ocean-subgraph`,
+      fetch
+    }),
+    cache: new InMemoryCache()
+  })
+
+  return client
+}
+
+export default function ApolloClientProvider({
+  children
+}: {
+  children: ReactNode
+}): ReactElement {
+  const { config } = useOcean()
+  const [client, setClient] = useState<ApolloClient<NormalizedCacheObject>>()
+
+  useEffect(() => {
+    if (!(config as ConfigHelperConfig)?.subgraphUri) return
+
+    const newClient = createClient((config as ConfigHelperConfig).subgraphUri)
+    setClient(newClient)
+  }, [config])
+
+  return client ? (
+    <ApolloProvider client={client}>{children}</ApolloProvider>
+  ) : (
+    <></>
```

Comment:
while in 5703ff5 I made sure `client` is present in all networks it seems like if this condition happens and there's no ApolloProvider the whole app breaks. We forgot to pass `children`

Reply:
so tricky, think we have to leave it like that as the best compromise. Basically whenever app encounters an undefined `config.subgraphUri` we prevent all children on `ApolloProvider` from rendering.

If I pass `children` in that case all the subcomponents using Apollo make app break. Since we're using hooks to access `ApolloProvider` there is no way in React to conditionally use them, unless we prevent every component from rendering whenever `ApolloProvider` is not present.

So best pragmatic way seems to make sure that `config.subgraphUri` is always defined which should happen successfully now in this PR. I'll log an error to console for all other cases

Reply:
check my comment above regarding how the app behaves when you don't have subgraph. The check is more or less to make ssr happy. After deployment even when the graph endpoint is broken the app works ok